### DISCUSSION TITLE
Recent update: change default distro to Jazzy ubuntu 24.04, add sonar costmap

### DISCUSSION THREAD

**Discussion by hippo5329** [hippo5329/linorobot2_hardware#25] [Announcements]
1. wiki: move software installation to [another wiki](https://github.com/hippo5329/micro_ros_arduino_examples_platformio/wiki#install-the-software-tools). micro_ros_agent is installed outside linorobot2_ws. As software tools and micro-ros agent can be used with the other projects. vscode/ssh remote development is described in that wiki.

2. Change default ros2 distro to the latest LTS Jazzy and Ubuntu 24.04. Jazzy is recommended for beginners and new designs.

3. Add sonar range sensor to global and local costmap.

Testing and comments are welcome.

**Comment by tbalatka** [comment]
With the change to Ubuntu 24.04 + ROS2 Jazzy for installing linorobot2_hardware, what version of ROS2 should one enter when installing linorobot2 (https://github.com/hippo5329/linorobot2)?

**Reply by hippo5329** [reply]
The versions of micro-ROS and ROS should match. So it is ROS2 Jazzy. Though default branches of linorobot2 and linorobot2_hardware should work for Foxy, Humble and Jazzy.
